src/shared/connection/passive_conn.cpp

```diff
@@ -0,0 +1,73 @@
+#include "passive_conn.hpp"
+#include "logger.hpp"
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#define QUEUE_SIZE 10
+
+bool PassiveConn::bind_and_listen(const std::string &t_address) {
+
+  setup(t_address);
+
+  auto address = get_address();
+
+  int res = bind(get_socket(), reinterpret_cast<struct sockaddr *>(&address),
+                 sizeof(address));
+  bool valid = is_valid(res, "Could not blind to socket.");
+
+  if (valid) {
+    listen(get_socket(), QUEUE_SIZE);
+    m_poll.add_socket(get_socket());
+    LOG_INFO("Listening on %d.", get_port())
+  }
+  return valid;
+}
+
+bool PassiveConn::accept_connection() {
+
+  // accepts only of there is an awaiting client trying to connect
+  // connection must also be set up.
+  if (!is_setup() || !m_poll.accept_awaits()) {
+    return false;
+  }
+
+  auto address = get_address();
+  socklen_t addr_len = sizeof(address);
+  int res = accept(get_socket(), reinterpret_cast<struct sockaddr *>(&address),
+                   &addr_len);
+
+  bool valid = is_valid(res, "Could not accept connection.");
+
+    if (valid) {
+      m_poll.add_socket(res); // add new connections to poll.
+  }
+
+  return valid;
+}
+
+bool PassiveConn::receive(std::string &t_data) {
+  m_client_socket = m_poll.select_socket_with_events();
+
+  bool valid = is_valid(m_client_socket, "Nothing to receive.");
```

Comment:
That seems like a spammy message, we can add another argument that sends a logger level

```
is_valid(m_client_socket, "Nothing to receive.", INFO);
```

Reply:
@khalilmasri  I am adding a DEBUG log to them so we can turn them on and off.